src/onnxflow/justbuildit/export.py

```diff
@@ -0,0 +1,614 @@
+            """
+            raise exp.StageError(msg)
+
+        return state
+
+
+class OptimizeOnnxModel(stage.Stage):
+    """
+    Stage that takes an ONNX file and uses ONNX Runtime to optimize it.
+    Important because this helps to perform constant folding, Redundant
+    node eliminations, Semantics-preserving node fusions
+
+    Expected inputs:
+     - state.intermediate_results contains a single .onnx file
+
+    Outputs:
+     - A *-opt.onnx file
+    """
+
+    def __init__(self):
+        super().__init__(
+            unique_name="optimize_onnx",
+            monitor_message="Optimizing ONNX file",
+        )
+
+    def fire(self, state: build.State):
+
+        # TODO: validate this input
+        # https://git.groq.io/code/Groq/-/issues/13947
+        input_onnx = state.intermediate_results[0]
+
+        # Perform some basic optimizations on the model to remove shape related
+        # information inserted for dynamic shape inference.
+        # Given that we're compiling against a fixed sequence length the dynamic
+        # shape information is not necessary
+        session_options = onnxruntime.SessionOptions()
+
+        # Set graph optimization level
+        session_options.graph_optimization_level = (
+            onnxruntime.GraphOptimizationLevel.ORT_ENABLE_BASIC
+        )
+
+        # To enable model serialization after graph optimization set this
+        session_options.optimized_model_filepath = state.opt_onnx_file
+
+        # Optimize graph
+        onnxruntime.InferenceSession(input_onnx, session_options)
+
+        # Check that the converted model is still valid
+        success_msg = "\tSuccess optimizing ONNX model"
+        fail_msg = "\tFailed optimizing ONNX model"
+        state.info.opt_onnx_exported = _check_model(
+            state.opt_onnx_file, success_msg, fail_msg
+        )
+
+        if state.info.opt_onnx_exported:
+            state.intermediate_results = [state.opt_onnx_file]
+        else:
+            msg = f"""
+            Unable to optimize ONNX file using ONNX runtime.
+            We recommend that you modify your model until it is
+            compatible with this third party software, then re-run benchit().
+            More information may be available in the log file at **{self.logfile_path}**
+            """
+            raise exp.StageError(msg)
+
+        return state
+
+
+class ConvertOnnxToFp16(stage.Stage):
+    """
+    Stage that takes an ONNX file and converts its trained parameters
+    to fp16.
+
+    Expected inputs:
+     - state.intermediate_results contains a single .onnx file
+
+    Outputs:
+     - A *-f16.onnx file with FP16 trained parameters
+    """
+
+    def __init__(self):
+        super().__init__(
+            unique_name="fp16_conversion",
+            monitor_message="Converting to FP16",
+        )
+
+    def fire(self, state: build.State):
+
+        # TODO: validate this input
+        # https://git.groq.io/code/Groq/-/issues/13947
```

Comment:
* Remove link to GroqFlow issue
* Replace with MLAgility issue
* Cross reference: Mention MLAgility issue in the GroqFlow issue